### DISCUSSION TITLE
Why airflow docker image comes with most of the providers preinstalled?

### DISCUSSION THREAD

**Discussion by Opalo** [apache/airflow#34674] [General] [answered]
Hi!

I'm working recently on refactoring quite big airflow repo and as a result want also make docker images smaller as well install only the dependencies that are indeed used/required in the codebase. Recently I popped into some providers dependencies that are not used in the codebase so I removed them (hive, postgres, jdbc, ssh and crypto). I built the image and the size did not change. So I run the image, then `pip install` and I see:
```
apache-airflow                           2.7.1
apache-airflow-providers-amazon          8.6.0
apache-airflow-providers-celery          3.3.3
apache-airflow-providers-cncf-kubernetes 7.5.0
apache-airflow-providers-common-sql      1.7.1
apache-airflow-providers-daskexecutor    1.0.1
apache-airflow-providers-docker          3.7.4
apache-airflow-providers-elasticsearch   5.0.1
apache-airflow-providers-ftp             3.5.1
apache-airflow-providers-google          10.7.0
apache-airflow-providers-grpc            3.2.2
apache-airflow-providers-hashicorp       3.4.2
apache-airflow-providers-http            4.5.1
apache-airflow-providers-imap            3.3.1
apache-airflow-providers-microsoft-azure 6.3.0
apache-airflow-providers-mysql           5.3.0
apache-airflow-providers-odbc            4.0.0
apache-airflow-providers-openlineage     1.0.2
apache-airflow-providers-postgres        5.6.0
apache-airflow-providers-redis           3.3.1
apache-airflow-providers-sendgrid        3.2.2
apache-airflow-providers-sftp            4.6.0
apache-airflow-providers-slack           8.0.0
apache-airflow-providers-snowflake       5.0.0
apache-airflow-providers-sqlite          3.4.3
apache-airflow-providers-ssh             3.7.2
```
Why is that? I don't want/need all these dependencies. Or do I? I don't to do any hacks with uninstalling them but why I can't just install the providers myself? Besides sth like `apache-airflow-providers-common-sql` which comes with `apache-airflow` and that's ok.

**Comment by o-nikolas** [comment, marked as answer]
We have multiple images available, it looks like you're using `apache/airflow:latest` but there is also `apache/airflow:slim-latest` which only contains core airflow. From [the docs](https://airflow.apache.org/docs/docker-stack/index.html):
> You can also use “slim” images that contain only core airflow and are about half the size of the “regular” images but you need to add all the [Reference for package extras](https://airflow.apache.org/docs/apache-airflow/stable/extra-packages-ref.html) and providers that you need separately via [Building the image](https://airflow.apache.org/docs/docker-stack/build.html#build-build-image).

I think the slim image is exactly what you're looking for! :smiley:

**Comment by Opalo** [comment]
Thanks @o-nikolas, that makes sense. Will definitely try it out.
